### DISCUSSION TITLE
libraryTarget and library

### DISCUSSION THREAD

**Discussion by AqilbekAbilaev** [webpack/webpack#14817] [Q&A]
output: {
        path: path.join(__dirname, 'dist'),
        filename: 'bundle.min.js',
        libraryTarget: 'var',
        library: 'Client'
    }, 
it is working perfect in webpack 4 but in 5 keeps appearing 

formHandler.js:6 Uncaught TypeError: Client.checkForName is not a function
    at Module.handleSubmit (formHandler.js:6)
    at HTMLInputElement.onclick (?input=:23)

**Comment by alexander-akait** [comment]
Please read docs https://webpack.js.org/configuration/output/#outputlibrarytarget, avoid usage `libraryTarget`, setup `library: { type: 'var', name: 'Client' }`

**Reply by AqilbekAbilaev** [reply]
Still problem :(

**Reply by alexander-akait** [reply]
Looks at output code by webpack and check what was changed, I think something wrong with export

**Reply by AqilbekAbilaev** [reply]
i am seeing with all my eyes nothing changed please help me if you do not object , I will upload source code you just npm install 😀

**Reply by AqilbekAbilaev** [reply]
Could you do me a favour please ?

**Comment by AqilbekAbilaev** [comment]
[fend-webpack-content.zip](https://github.com/webpack/webpack/files/7597069/fend-webpack-content.zip)

**Reply by alexander-akait** [reply]
You should not use `library` in your case, remove it, and you should avoid using code from bundle inside `HTML` file, write code inside `index.js` and use `addEventListener` to attach, yes we can fix your case, but it is wrong and bad idea, better do not use bad approaches.

Anyway:
- Set:
```
 output: {
        library: {
            type: 'window',
            name: 'Client'
        }
    },
```
2. Fix you HTML, i.e. always use `Client` before methods (i.e. Client.handleSubmit and etc)
3. Fix your JS

```
// You should import your code, not try to using it through `Client`
import { checkForname } from "./nameChecker";

function handleSubmit(event) {
    event.preventDefault()

    // check what text was put into the form field
    let formText = document.getElementById('name').value
    checkForname(formText)

    console.log("::: Form Submitted :::")
    fetch('http://localhost:8081/test')
    .then(res => res.json())
    .then(function(res) {
        document.getElementById('results').innerHTML = res.message
    })
}

export { handleSubmit }
```

Shorty - avoid usage your code through global variable, it is bad idea

**Reply by AqilbekAbilaev** [reply]
Wow bro it worked 😍👏👏 ok I follow your instructions. Actually it is not my code it is from the course. They used Webpack 4 during the course, But I want to be up-to-date and want to try in Webpack 5. The reason is clear. Thank you so much!

**Reply by AqilbekAbilaev** [reply]
I am sorry. But here one problem appeared we directly imported checkForname in handleSubmit.js but what if we have a lot of js files. Actually this job should be done by index.js but it is not doing his job. How to fix it...? One more question which Webpack 4 or 5 would you recommend me to use ? Thank you in advance Спасибо за все :)

**Reply by alexander-akait** [reply]
> what if we have a lot of js files

You should do not worry about it and have `imports` how much do you want, webpack was created for this, you will not have duplicated modules in your bundle. 

> One more question which Webpack 4 or 5 would you recommend me to use

webpack v5
